software/webapp/src/main/java/brooklyn/entity/webapp/JavaWebAppSshDriver.java

```diff
@@ -89,11 +89,11 @@ protected String inferRootUrl() {
         if (isProtocolEnabled("https")) {
             Integer port = getHttpsPort();
             checkNotNull(port, "HTTPS_PORT sensors not set; is an acceptable port available?");
-            return String.format("https://%s:%s/", getHostname(), port);
+            return String.format("https://%s:%s/", getSubnetHostname(), port);
```

Comment:
Am I right that `getSubnetHostname()` will return the same as `getHostname()` if not running in Clocker or a `SubnetTier`?

What about for AWS - will the subnetHostname still be the public hostname, rather than the private IP of the VM?

Assuming so, this looks sensible.